### PR TITLE
Update the default values of dynamic config to not depend on static config

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**

Update the default values of dynamic config to be static values, and to not depend on values read from static config file

**Why?**

- Dynamic config and static config should be independent of each other
- With this change we can create a source of truth to store the default values of dynamic config

**How did you test it?**
existing test

**Potential risks**

**Release notes**
If there is no `system.grpcMaxSizeInByte` config, it needs to be created. And the value should be the same as the `maxPayloadSize` in the static config.

**Documentation Changes**

### CODE REVIEW

File: common/archiver/archivalMetadata.go [hunk 1]
```diff
@@ -177,7 +183,7 @@ func (a *archivalConfig) ReadEnabled() bool {
 	if !a.ClusterConfiguredForArchival() {
 		return false
 	}
-	return a.enableRead()
+	return a.staticEnableRead && a.dynamicEnableRead()
```
Comment (davidporter-id-au) [inline common/archiver/archivalMetadata.go:186]: This configuration might be surprising as some folks might expect either `||` or `if dynamicConfig != nil return dynamicConfig else staticConfig`

Comment (davidporter-id-au) [reply common/archiver/archivalMetadata.go:186]: I *dont* have a strong opinion on what it should be as I don't know enough about how this and similar config values are used in production, but I think maybe we should call out this both in the head of the diff and maybe in comments?

File: common/archiver/archivalMetadata.go [hunk 2]
```diff
@@ -55,7 +55,8 @@ type (
 	archivalConfig struct {
 		staticClusterStatus  ArchivalStatus
 		dynamicClusterStatus dynamicconfig.StringPropertyFn
-		enableRead           dynamicconfig.BoolPropertyFn
+		staticEnableRead     bool
```
[the remark this answers is not in the document]
Comment (Shaddoll) [reply common/archiver/archivalMetadata.go:58]: yes, it's hided through `ReadEnabled() bool`. if both are true, read is enabled

File: common/archiver/archivalMetadata.go (hunk 1 above)
Comment (Shaddoll) [reply common/archiver/archivalMetadata.go:186]: This is to keep the behavior the same as GetClusterStatus function

**Review by davidporter-id-au — APPROVED** [review]
Looks good. I'm mildly worried this might manifest in some behaviour changes, but maybe we can use Steven's config dump to detect this.

File: client/clientfactory.go [hunk 3]
```diff
@@ -117,7 +117,7 @@ func (cf *rpcClientFactory) NewHistoryClientWithTimeout(timeout time.Duration) (
 	peerResolver := history.NewPeerResolver(cf.numberOfHistoryShards, cf.resolver, namedPort)
 
 	supportedMessageSize := cf.rpcFactory.GetMaxMessageSize()
-	maxSizeConfig := cf.dynConfig.GetIntProperty(dynamicconfig.GRPCMaxSizeInByte, supportedMessageSize)
+	maxSizeConfig := cf.dynConfig.GetIntProperty(dynamicconfig.GRPCMaxSizeInByte, 4*1024*1024)
```
[the remark this answers is not in the document]
Comment (Shaddoll) [reply client/clientfactory.go:120]: will be addressed in https://github.com/uber/cadence/pull/4863

**Pull request merged**